mobile/lib/components/ra_bottom_navigation_bar.dart

```diff
@@ -0,0 +1,123 @@
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+import 'package:radioaktywne/extensions/build_context.dart';
+import 'package:radioaktywne/router/ra_routes.dart';
+
+/// Represents aplication's bottom navigation bar.
+class RaBottomNavigationBar extends StatelessWidget {
+  const RaBottomNavigationBar({
+    super.key,
+    required this.currentPath,
+    this.borderWidth = 5,
+    this.onNavigate,
+  });
+
+  /// Name of the current page provided by
+  /// the [GoRouter]'s state.
+  final String currentPath;
+
+  /// The width of this widget's green border.
+  final double borderWidth;
+
+  /// Optional function to be called on every
+  /// navigation to a route __different__ than
+  /// the current one.
+  final void Function()? onNavigate;
+
+  // Leaving this as a reminder...
+  // static const labels = <String>['home', 'mic', 'album', 'article'];
+  // static const widths = <double>[30, 21, 32.5, 27];
+  // static const heights = <double>[26, 28.5, 32.5, 27];
+
+  static const _itemInfo = [
+    (
+      pagePath: RaRoutes.home,
+      icon: Icons.home_outlined,
+      activeIcon: Icons.home,
+      iconSize: 30.0,
+    ),
+    (
+      pagePath: RaRoutes.recordings,
+      icon: Icons.mic_none_outlined,
+      activeIcon: Icons.mic,
+      iconSize: 28.5,
+    ),
+    (
+      pagePath: RaRoutes.albumOfTheWeek,
+      icon: Icons.album_outlined,
+      activeIcon: Icons.album,
+      iconSize: 32.5,
+    ),
+    (
+      pagePath: RaRoutes.articles,
+      icon: Icons.article_outlined,
+      activeIcon: Icons.article,
+      iconSize: 27.0,
+    ),
+  ];
+
+  List<Widget> _makeList() {
+    return _itemInfo
+        .map(
+          (item) => RaBottomNavigationBarItem(
+            activeIcon: Icon(item.activeIcon),
+            icon: Icon(item.icon),
+            iconSize: item.iconSize,
+            pagePath: item.pagePath,
+            isSelected: item.pagePath == currentPath,
+            onNavigate: onNavigate,
+          ),
+        )
+        .toList();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.only(top: borderWidth),
+      color: context.colors.highlightGreen,
+      child: Container(
+        // padding: const EdgeInsets.symmetric(horizontal: 5),
+        color: context.colors.backgroundDark,
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+          children: _makeList(),
+        ),
+      ),
+    );
+  }
+}
```

Comment:
Unfortunately, right now the bottom navigation bar is outside of the safe area. I checked only on an iPhone but I suppose it will be the case on other phones as well. My suggestion would be to wrap the container in a `SafeArea` widget and then again wrap it inside the `Container` widget so it is possible to change the color to background dark.

```dart
@override
Widget build(BuildContext context) {
  return Container(
    color: context.colors.backgroundDark,
    child: SafeArea(
      child: Container(
        padding: EdgeInsets.only(top: borderWidth),
        color: context.colors.highlightGreen,
        child: Container(
          // padding: const EdgeInsets.symmetric(horizontal: 5),
          color: context.colors.backgroundDark,
          child: Row(
            mainAxisAlignment: MainAxisAlignment.spaceEvenly,
            children: _makeList(),
          ),
        ),
      ),
    ),
  );
}
```

<img src="https://github.com/RadioAktywne/ra-app/assets/101600853/527730c1-a3a0-4d3f-8be3-1ad2d3557a0f" width="300">